storage/innobase/fil/fil0fil.cc

```diff
@@ -11872,6 +11872,11 @@ dberr_t fil_open_for_xtrabackup(const std::string &path,
 
   ut_a(space != NULL);
 
+  char *fn = fil_node_create(file.filepath(), n_pages, space, false, false);
+  if (fn == nullptr) {
+    return (DB_ERROR);
+  }
+
```

Comment:
any reason to move this code up? I see that fil_space_t* is also available later too

Reply:
@satya-bodapati as part of b184bd3 a new assertion was added in get_space_by_id_from_map - https://github.com/mysql/mysql-server/blob/mysql-8.0.28/storage/innobase/fil/fil0fil.cc#L736 . Now the fil_space_t requires to have a file attached to it.

A few lines below, we will call fsp_flags_set_encryption, which calls Fil_shard::get_space_by_id which calls Fil_shard::get_space_by_id_from_map, resulting on failing at the new assertion. 

calling fil_node_create to attach the file to the tablespace earlier fixes the issue.